FrameworkSystem/DB/ProxyDB.py

```diff
@@ -595,35 +677,49 @@ def getVOMSProxy( self, userDN, userGroup, requiredLifeTime = False, requestedVO
           if requiredLifeTime and requiredLifeTime <= vomsTime and requiredLifeTime <= remainingSecs:
             return S_OK( ( chain, min( vomsTime, remainingSecs ) ) )
 
-    retVal = self.getProxy( userDN, userGroup, requiredLifeTime )
-    if not retVal[ 'OK' ]:
-      return retVal
-    chain, secsLeft = retVal[ 'Value' ]
+    if self.__isPUSPdn( userDN ):
+      # Get the Per User SubProxy if one is requested
+      result = self.__getPUSProxy( userDN, userGroup, requiredLifeTime, requestedVOMSAttr )
+      if not result['OK']:
+        return result
+      pemData = result[ 'Value' ][0]
+      chain = X509Chain()
+      result = chain.loadProxyFromString( pemData )
+      if not result[ 'OK' ]:
+        return result
+    else:
+      # Get the stored proxy and dress it with the VOMS extension
+      retVal = self.getProxy( userDN, userGroup, requiredLifeTime )
+      if not retVal[ 'OK' ]:
+        return retVal
+      chain, secsLeft = retVal[ 'Value' ]
 
-    if requiredLifeTime and requiredLifeTime > secsLeft:
-      return S_ERROR( "Stored proxy is not long lived enough" )
+      if requiredLifeTime and requiredLifeTime > secsLeft:
+        return S_ERROR( "Stored proxy is not long lived enough" )
 
-    vomsMgr = VOMS()
+      vomsMgr = VOMS()
 
-    retVal = vomsMgr.getVOMSAttributes( chain )
-    if retVal[ 'OK' ]:
-      attrs = retVal[ 'Value' ]
-      if len( attrs ) > 0:
-        if attrs[0] != vomsAttr:
-          return S_ERROR( "Stored proxy has already a different VOMS attribute %s than requested %s" % ( vomsAttr, attrs[0] ) )
-        else:
-          result = self.__storeVOMSProxy( userDN, userGroup, vomsAttr, chain )
-          if not result[ 'OK' ]:
-            return result
-          secsLeft = result[ 'Value' ]
-          if requiredLifeTime and requiredLifeTime <= secsLeft:
-            return S_OK( ( chain, secsLeft ) )
-          return S_ERROR( "Stored proxy has already a different VOMS attribute and is not long lived enough" )
+      retVal = vomsMgr.getVOMSAttributes( chain )
+      if retVal[ 'OK' ]:
+        attrs = retVal[ 'Value' ]
+        if len( attrs ) > 0:
```

Comment:
This is the old code, yes can be changed. I prefer not to make trivial beautifications which are rather a matter of taste than any other reason